From b75196ac27fb14f03c0257a6eadb375947ade815 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 24 Feb 2014 15:28:58 +0100
Subject: [PATCH] network/parse-ifname.sh: define parse_ifname_opts()

if no "ifname=" parameter was given on the kernel command line, and only
later injected via cmdline.d (e.g. with iBFT), the parse_ifname_opts()
function would not be defined in ifname-genrules.sh .
---
 modules.d/40network/parse-ifname.sh | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/modules.d/40network/parse-ifname.sh b/modules.d/40network/parse-ifname.sh
index 9632680..9cc03a9 100755
--- a/modules.d/40network/parse-ifname.sh
+++ b/modules.d/40network/parse-ifname.sh
@@ -10,11 +10,6 @@
 # Note when using ifname= to get persistent interface names, you must specify
 # an ifname= argument for each interface used in an ip= or fcoe= argument
 
-# check if there are any ifname parameters
-if ! getarg ifname= >/dev/null ; then
-    return
-fi
-
 parse_ifname_opts() {
     local IFS=:
     set $1
@@ -31,6 +26,11 @@ parse_ifname_opts() {
     esac
 }
 
+# check if there are any ifname parameters
+if ! getarg ifname= >/dev/null ; then
+    return
+fi
+
 # Check ifname= lines
 for p in $(getargs ifname=); do
     parse_ifname_opts $p
